app/scripts/migrations/039.js

```diff
@@ -0,0 +1,66 @@
+const version = 39
+const clone = require('clone')
+const ethUtil = require('ethereumjs-util')
+
+const DAI_V1_CONTRACT_ADDRESS = '0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359'
+const DAI_V1_TOKEN_SYMBOL = 'DAI'
+const SAI_TOKEN_SYMBOL = 'SAI'
+
+function isOldDai (token = {}) {
+  return typeof token === 'object' && token.symbol === DAI_V1_TOKEN_SYMBOL &&
```

Comment:
Nit: technically this doesn't guard against the token being `null`. A null parameter will blow up here.

Reply:
Good catch, I fixed this with a branch in the loop below and added a test

Reply:
I'm having trouble locating the test you added; also `isOldDai` is used in multiple places, and you didn't add a guard in the first one (line 39). Maybe best moving the null check into this function?

Reply:
I added another `null` to the "invalid values" test case. I can add another one for the `tokens` property too, that's a good point.

Reply:
Added more tests for various things